Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -438,36 +440,44 @@ private static void MergeDictionary(IList<Dictionary<string, FlexMatch>> mergeFr
 
         private void RunMatchExpression(FlexMatch binary64DecodedMatch, AnalyzeContext context, MatchExpression matchExpression)
         {
-            if (matchExpression.IntrafileRegexes?.Count > 0)
+            if (!string.IsNullOrEmpty(matchExpression.ContentsRegex))
             {
-                Debug.Assert(binary64DecodedMatch == null, "Decoded binary64 should not be null");
-                RunMatchExpressionForIntrafileRegexes(context, matchExpression);
+                RunMatchExpressionForContentsRegex(binary64DecodedMatch, context, matchExpression);
+                return;
             }
-            else if (!string.IsNullOrEmpty(matchExpression.ContentsRegex))
+
+            Debug.Assert(binary64DecodedMatch == null, "Decoded binary64 should be null");
+
+            if (matchExpression.IntrafileRegexes?.Count > 0 ||
+                matchExpression.SingleLineRegexes?.Count > 0)
             {
-                RunMatchExpressionForContentsRegex(binary64DecodedMatch, context, matchExpression);
+                RunMatchExpressionForSingleLineAndIntrafileRegexes(context, matchExpression);
             }
             else if (!string.IsNullOrEmpty(matchExpression.FileNameAllowRegex))
             {
-                Debug.Assert(binary64DecodedMatch == null, "Decoded binary64 should not be null");
                 RunMatchExpressionForFileNameRegex(context, matchExpression);
             }
             else
             {
-                // Both FileNameAllowRegex and ContentRegex are null or empty.
+                throw new InvalidOperationException("Malformed expression contains no regexes.");
             }
         }
 
+        private void RunMatchExpressionForSingleLineAndIntrafileRegexes(AnalyzeContext context, MatchExpression matchExpression)
+        {
+            RunMatchExpressionForIntrafileRegexes(context, matchExpression);
```

Comment:
>RunMatchExpressionForIntrafileRegexes [](http://example.com/codeflow?start=12&length=37)

is this intentional? with this, we would be able to run Intrafile and SingleLine

Reply:
Yes. What problem do you see? Currently, you can create a single rule that has both Intrafile and SingleFile regexes. Do you think we should not permit that??

Reply:
i thought we were doing something like one or the other. :)

but, doing both will permit a lot of combinations.
it was just to confirm